Data/Aeson/Encode/Pretty.hs

```diff
@@ -62,21 +62,26 @@ import Data.List (intersperse, sortBy, elemIndex)
 import Data.Maybe (fromMaybe)
 import Data.Monoid ((<>), mconcat, mempty)
 import Data.Ord
+import Data.Scientific
 import Data.Text (Text)
 import Data.Text.Lazy.Builder (Builder, toLazyText)
+import Data.Text.Lazy.Builder.Scientific
 import Data.Text.Lazy.Encoding (encodeUtf8)
 import qualified Data.Vector as V (toList)
 
 data PState = PState { pstIndent :: Int
-                     , pstLevel  :: Int
-                     , pstSort   :: [(Text, Value)] -> [(Text, Value)]
+                     , pstLevel :: Int
+                     , pstSort :: [(Text, Value)] -> [(Text, Value)]
+                     , pstFPFormat :: FPFormat
```

Comment:
It would be more generally useful to be able to supply a `Scientific -> Value` function here.